pkg/scheduler/apis/config/v1/defaults_test.go

```diff
@@ -144,14 +145,15 @@ func TestSchedulerDefaults(t *testing.T) {
 					Burst:       100,
 					ContentType: "application/vnd.kubernetes.protobuf",
 				},
-				PercentageOfNodesToScore: pointer.Int32Ptr(0),
-				PodInitialBackoffSeconds: pointer.Int64Ptr(1),
-				PodMaxBackoffSeconds:     pointer.Int64Ptr(10),
+				PercentageOfNodesToScore: pointer.Int32(config.DefaultPercentageOfNodesToScore),
+				PodInitialBackoffSeconds: pointer.Int64(1),
+				PodMaxBackoffSeconds:     pointer.Int64(10),
 				Profiles: []configv1.KubeSchedulerProfile{
 					{
-						Plugins:       getDefaultPlugins(),
-						PluginConfig:  pluginConfigs,
-						SchedulerName: pointer.StringPtr("default-scheduler"),
+						Plugins:                  getDefaultPlugins(),
+						PluginConfig:             pluginConfigs,
+						SchedulerName:            pointer.StringPtr("default-scheduler"),
+						PercentageOfNodesToScore: nil,
```

Comment:
No, it's needed. Otherwise, we would get ` Got unexpected defaults (-want, +got):`  error.

Reply:
> Maybe we should do two things, but on the nit side:
> 
> * Only set the default top level percentage if there is at least one profile that didn't set it up.
> * Validate that if all the profiles set percentage, then the global one should be nil.
> 
> This would prevent confusion when a user sets both values and then it's a bit unclear which one takes effect.

In a multi-profile scenario, I'm afraid this might cause more confusion. If a profile sets its percentage while another one didn't, what value will be used for the second profile? We should clearly document it if a profile percentage is set, it will be used. Otherwise, the global one (the default value) will be used.

Reply:
> are you sure you didn't get confused with errors for the global field?

I know that's the global general error.  Without setting default plugins and other profile parameters, the expected profile empty {} won't match the default config.

```
            - 		{},
            + 		{
            + 			SchedulerName: &"default-scheduler",
            + 			Plugins:       &v1.Plugins{MultiPoint: v1.PluginSet{Enabled: []v1.Plugin{...}}},
            + 			PluginConfig: []v1.PluginConfig{
            + 				{Name: "DefaultPreemption", Args: runtime.RawExtension{...}},
            + 				{Name: "InterPodAffinity", Args: runtime.RawExtension{...}},
            + 				{Name: "NodeAffinity", Args: runtime.RawExtension{...}},
            + 				{Name: "NodeResourcesBalancedAllocation", Args: runtime.RawExtension{...}},
            + 				{Name: "NodeResourcesFit", Args: runtime.RawExtension{...}},
            + 				{Name: "PodTopologySpread", Args: runtime.RawExtension{...}},
            + 				{Name: "VolumeBinding", Args: runtime.RawExtension{...}},
            + 			},
            + 		},
              	},
```